pkg/application/types_test.go

```diff
@@ -132,3 +132,176 @@ func TestClusterValuesNodePool(t *testing.T) {
 		})
 	}
 }
+
+func TestClusterValues(t *testing.T) {
+	type errorTestCases struct {
+		description string
+		input       string
+		expectError bool
+		expected    ClusterValues
+	}
+
+	for _, scenario := range []errorTestCases{
+		{
+			description: "empty values",
+			input:       ``,
+			expectError: false,
+			expected:    ClusterValues{},
+		},
+		{
+			description: "old schema",
+			input: `baseDomain: "foo.com"
+controlPlane:
+  replicas: 3`,
+			expectError: false,
+			expected: ClusterValues{
+				BaseDomain: "foo.com",
+				ControlPlane: ControlPlane{
+					Replicas: 3,
+				},
+			},
+		},
+		{
+			description: "old schema with nodepools map",
+			input: `baseDomain: "foo.com"
+controlPlane:
```

Comment:
Formatting looks off here. Is this right?

Reply:
This is correct. It's a raw yaml string so the indentation is important (and needs to use spaces instead of tabs)